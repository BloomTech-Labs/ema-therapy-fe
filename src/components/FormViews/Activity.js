import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ActivityTag from './ActivityTag';

const Activity = ({ activityType, addActivities, isSelectedForEdit }) => {
  return (
    <>
      <ActivityButtonToggle>
        <ActivityTag
          isSelectedForEdit={isSelectedForEdit}
          activityType={activityType}
          addActivities={addActivities}
        >
          {activityType}
        </ActivityTag>
      </ActivityButtonToggle>
    </>
  );
};

Activity.propTypes = {
  activityType: PropTypes.string.isRequired,
  addActivities: PropTypes.func.isRequired,
  isSelectedForEdit: PropTypes.bool.isRequired,
};

const ActivityButtonToggle = styled.div`
  display: inline-block;
`;

export default Activity;
