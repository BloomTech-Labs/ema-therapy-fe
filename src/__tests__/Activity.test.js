import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Activity from '../components/FormViews/Activity';
import 'jest-styled-components';

describe('Activity component', () => {
  test('renders correctly', () => {
    expect(
      render(
        <Activity
          addActivities={() => {}}
          activityType={{ icon: 'mockIcon', type: 'mockType' }}
        />,
      ),
    ).toMatchSnapshot();
  });
  test('handles click', () => {
    const addActivitiesMock = jest.fn();
    const activityTypeMock = { icon: 'mockIcon', type: 'mockType' };
    const { getByTestId } = render(
      <Activity
        addActivities={addActivitiesMock}
        activityType={activityTypeMock}
      />,
    );
    fireEvent.click(getByTestId('toggle'));
    expect(addActivitiesMock).toHaveBeenCalled();
  });
});
