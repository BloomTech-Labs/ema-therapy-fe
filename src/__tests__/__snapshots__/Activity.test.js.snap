// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  border: 1px solid black;
  border-radius: 100%;
  padding: 4px;
}

.c0 {
  padding: 5px;
  width: 70px;
  border: none;
  background: none;
  margin-right: 3px;
  margin-top: 27px;
  color: limegreen;
}

<body>
    <div>
      <button
        class="c0"
        data-testid="toggle"
        type="button"
      >
        <svg
          class="c1"
          height="30"
          viewBox="0 0 500 500"
          width="30"
        >
          <path
            d="mockIcon"
          />
        </svg>
        <p>
          mockType
        </p>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bwzfXH sc-htpNat enBhE"
      data-testid="toggle"
      type="button"
    >
      <svg
        class="sc-bdVaJa hEIfhB"
        height="30"
        viewBox="0 0 500 500"
        width="30"
      >
        <path
          d="mockIcon"
        />
      </svg>
      <p>
        mockType
      </p>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
